### PR TITLE
Feat/new bottom nav

### PR DESCRIPTION
**Pull request by rBergCode**
# Summary
The objective was to have a new menu that is easier to use. For this I changed the base menu into a bottom bar menu and changed the bottom bar menu from the saves hike screen into tabs that can be scroll between.

# Details
I also enabled the edge to edge property in order to fill in the device screen.

### CODE REVIEW

File: app/src/main/java/ch/hikemate/app/MainActivity.kt [hunk 1]
```diff
@@ -39,7 +45,21 @@ class MainActivity : ComponentActivity() {
...
+      HikeMateTheme {
+        val systemBarStyle by remember {
+          val defaultSystemBarColor = android.graphics.Color.TRANSPARENT
+          mutableStateOf(
+              SystemBarStyle.auto(
+                  lightScrim = defaultSystemBarColor, darkScrim = defaultSystemBarColor))
+        }
+        LaunchedEffect(systemBarStyle) {
+          enableEdgeToEdge(statusBarStyle = systemBarStyle, navigationBarStyle = systemBarStyle)
+        }
```
Comment (Wylwi) [inline app/src/main/java/ch/hikemate/app/MainActivity.kt:59]: Nitpick: You should mention this changes in your PR description

File: app/src/main/java/ch/hikemate/app/ui/auth/CreateAccountScreen.kt [hunk 2]
```diff
@@ -109,10 +109,10 @@ fun CreateAccountScreen(navigationActions: NavigationActions, authViewModel: Aut
       modifier =
           Modifier.testTag(Screen.CREATE_ACCOUNT)
               .padding(
-                  // Add padding to the sidebar padding
+                  // Add padding to the top padding
```
Comment (Wylwi) [inline app/src/main/java/ch/hikemate/app/ui/auth/CreateAccountScreen.kt:112]: Nitpick: I don't understand this comment. Could you rephrase it? (Multiple occurrences of the word padding)

File: app/src/main/java/ch/hikemate/app/ui/auth/SignInWithEmailScreen.kt [hunk 3]
```diff
@@ -78,10 +78,10 @@ fun SignInWithEmailScreen(navigationActions: NavigationActions, authViewModel: A
       modifier =
           Modifier.testTag(Screen.SIGN_IN_WITH_EMAIL)
               .padding(
-                  // Add padding to the sidebar padding
+                  // Add padding to the top padding
```
Comment (Wylwi) [inline app/src/main/java/ch/hikemate/app/ui/auth/SignInWithEmailScreen.kt:81]: ditto

File: app/src/main/java/ch/hikemate/app/MainActivity.kt (hunk 1 above)
Comment (rBergCode) [reply app/src/main/java/ch/hikemate/app/MainActivity.kt:59]: You're right. Fixed!

File: app/src/main/java/ch/hikemate/app/ui/navigation/BottomBarNavigation.kt [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.semantics.SemanticsPropertyKey
+import androidx.compose.ui.semantics.semantics
+
+const val TEST_TAG_BOTTOM_BAR = "TEST_TAG_BOTTOM_BAR"
+const val TEST_TAG_MENU_ITEM_PREFIX = "TEST_TAG_MENU_ITEM_"
+
+val IsSelectedKey = SemanticsPropertyKey<Boolean>("IsSelected")
```
Comment (Wylwi) [inline app/src/main/java/ch/hikemate/app/ui/navigation/BottomBarNavigation.kt:18]: Question: What is a SemanticsPropertyKey? From what I read, it's something related to tests. Is it so? If so, could you write a small comment, please?

File: app/src/main/java/ch/hikemate/app/ui/navigation/BottomBarNavigation.kt [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        NavigationBar(
+            modifier = Modifier.testTag(TEST_TAG_BOTTOM_BAR),
+        ) {
+          tabList.forEach { tab ->
+            NavigationBarItem(
+                icon = {
+                  Icon(
+                      tab.icon,
+                      contentDescription = tab.textId,
+                      modifier = Modifier.testTag(TEST_TAG_MENU_ITEM_PREFIX + tab.route + "Icon"),
```
Comment (Wylwi) [inline app/src/main/java/ch/hikemate/app/ui/navigation/BottomBarNavigation.kt:47]: Nitpick: Could you create a constant for the "Icon" string?

File: app/src/main/java/ch/hikemate/app/ui/navigation/BottomBarNavigation.kt [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+                  )
+                },
+                label = { Text(text = tab.textId) },
+                selected = tab.route == selectedItem,
+                onClick = { onTabSelect(tab) },
+                modifier =
+                    Modifier.testTag(TEST_TAG_MENU_ITEM_PREFIX + tab.route).semantics {
+                      set(IsSelectedKey, tab.route == selectedItem)
+                    },
```
Comment (Wylwi) [inline app/src/main/java/ch/hikemate/app/ui/navigation/BottomBarNavigation.kt:56]: Request: Could you explain (here and/or using a comment) what the `.semantics { .... }` part do? I'm struggling to understand why this is required

File: app/src/main/java/ch/hikemate/app/ui/profile/EditProfileScreen.kt [hunk 7]
```diff
@@ -76,7 +76,7 @@ fun EditProfileScreen(
       modifier =
           Modifier.testTag(Screen.EDIT_PROFILE)
               .padding(
-                  // Add padding to the sidebar padding
+                  // Add padding to the top padding
```
Comment (Wylwi) [inline app/src/main/java/ch/hikemate/app/ui/profile/EditProfileScreen.kt:79]: ditto

File: app/src/main/java/ch/hikemate/app/ui/saved/SavedHikesScreen.kt [hunk 8]
```diff
@@ -184,6 +202,6 @@ private fun SavedHikesBottomMenu(
  * element of the enum will be the left-most section in the bottom menu.
  */
 enum class SavedHikesScreen(val label: String, @DrawableRes val icon: Int) {
-  Planned("Planned", R.drawable.calendar_today),
-  Saved("Saved", R.drawable.bookmark)
+  Saved("Saved", R.drawable.bookmark),
+  Planned("Planned", R.drawable.calendar_today)
```
Comment (Wylwi) [inline app/src/main/java/ch/hikemate/app/ui/saved/SavedHikesScreen.kt:206]: Question: Am I missing something here, or is it only a position change?

File: app/src/main/java/ch/hikemate/app/ui/saved/SavedHikesScreen.kt [hunk 9]
```diff
@@ -52,27 +55,43 @@ fun SavedHikesScreen(
...
-          SavedHikesScreen.Saved -> SavedHikes(savedHikes)
+      // Navigation items between nearby hikes, planned hikes, and saved hikes
+      SavedHikesTabsMenu(
+          selectedIndex = currentSection.ordinal, onSelectedChange = { currentSection = it })
+
+      HorizontalPager(
+          state = pagerState,
+          modifier = Modifier.fillMaxWidth().weight(1f),
+      ) { pageIndex ->
+        Column(modifier = Modifier.weight(1f).testTag(TEST_TAG_SAVED_HIKES_SECTION_CONTAINER)) {
```
Comment (Wylwi) [inline app/src/main/java/ch/hikemate/app/ui/saved/SavedHikesScreen.kt:86]: Question: Why is a weight necessary here?

**Review by Wylwi — COMMENTED** [review]
Just got a few questions, otherwise, amazing (and I assume time-consuming) job!

File: app/src/main/java/ch/hikemate/app/ui/navigation/BottomBarNavigation.kt (hunk 4 above)
Comment (rBergCode) [reply app/src/main/java/ch/hikemate/app/ui/navigation/BottomBarNavigation.kt:18]: I removed it. Forgot to do it.

File: app/src/main/java/ch/hikemate/app/ui/navigation/BottomBarNavigation.kt (hunk 6 above)
Comment (rBergCode) [reply app/src/main/java/ch/hikemate/app/ui/navigation/BottomBarNavigation.kt:56]: Same as above.

File: app/src/main/java/ch/hikemate/app/ui/saved/SavedHikesScreen.kt (hunk 8 above)
Comment (rBergCode) [reply app/src/main/java/ch/hikemate/app/ui/saved/SavedHikesScreen.kt:206]: It is only a position change to reflect the Figma.

File: app/src/main/java/ch/hikemate/app/ui/navigation/BottomBarNavigation.kt (hunk 5 above)
Comment (rBergCode) [reply app/src/main/java/ch/hikemate/app/ui/navigation/BottomBarNavigation.kt:47]: Actually, this was useless! I removed the modifier.

File: app/src/main/java/ch/hikemate/app/ui/saved/SavedHikesScreen.kt (hunk 9 above)
Comment (rBergCode) [reply app/src/main/java/ch/hikemate/app/ui/saved/SavedHikesScreen.kt:86]: It is not. Thanks!

File: app/src/main/java/ch/hikemate/app/ui/saved/SavedHikesScreen.kt [hunk 10]
```diff
@@ -62,41 +71,55 @@ fun SavedHikesScreen(
...
+            errorMessageId != null ->
+                CenteredErrorAction(
+                    errorMessage = errorMessageId!!,
+                    actionIcon = Icons.Default.Refresh,
+                    actionContentDescription = R.string.saved_hikes_screen_refresh_button_action,
+                    onAction = { savedHikesViewModel.loadSavedHikes() })
+            else ->
+                HorizontalPager(
+                    state = pagerState,
+                    modifier = Modifier.fillMaxWidth().weight(1f),
```
Comment (Wylwi) [inline app/src/main/java/ch/hikemate/app/ui/saved/SavedHikesScreen.kt:110]: Question: You removed the other weight, but not this one. Is it an omission or required?

**Review by Wylwi — COMMENTED** [review]
Just a last question about a previous question! Otherwise, good to go!

File: app/src/main/java/ch/hikemate/app/ui/saved/SavedHikesScreen.kt (hunk 10 above)
Comment (rBergCode) [reply app/src/main/java/ch/hikemate/app/ui/saved/SavedHikesScreen.kt]: It's an omission also.

**Review by Wylwi — APPROVED** [review]
Thanks for the changes! Looks good to me!
